src/ns/core/notifier.py

```diff
@@ -74,18 +74,17 @@ def _create_notification_record(
         hash_sum = sha256(notification.model_dump_json().encode("utf-8")).hexdigest()
         return models.NotificationRecord(hash_sum=hash_sum, sent=False)
 
-    async def _ensure_not_sent(self, *, hash_sum: str) -> bool:
-        """Ensures that the notification has not been sent already.
+    async def _check_if_sent(self, *, hash_sum: str) -> bool:
```

Comment:
```suggestion
    async def _has_been_sent(self, *, hash_sum: str) -> bool:
```
Would slightly change this, so it's clear what the expected positive outcome is